### PR TITLE
Pyae linn(anagram checker)

### LINKED ISSUE #78: PyaeLinn(anagram_checker)

### PR DESCRIPTION
**Pull request by PyaeLinn01**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations
- [x] The function is not called in the function file

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] There are no commented lines of code
- [x] There are no `print` statements anywhere
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

### CODE REVIEW

**Comment by PyaeLinn01** [comment]
I think i passe all checks. Please review and mark for the template and help me merge to the main branch

**Comment by Akan186** [comment]
I think "MayMon-T3807" is reviewing your code. She has permission to check and allow the merge.

**Comment by MayMon-T3807** [comment]
> I think "MayMon-T3807" is reviewing your code. She has permission to check and allow the merge.

yes Now, I am reviewing. Thank you.

**Comment by PyaeLinn01** [comment]
Thank you guys.

**Comment by Akan186** [comment]
I removed "MayMon-T3807" by mistake. Sorry.

File: solutions/anagram_checker.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    """
+    if not isinstance(str1, str) or not isinstance(str2, str):
+        raise TypeError("Both inputs must be strings.")
+
+    # Normalize by removing spaces and converting to lowercase
+    normalized_str1 = str1.replace(" ", "").lower()
+    normalized_str2 = str2.replace(" ", "").lower()
+
+    # Check if sorted characters match
+    return sorted(normalized_str1) == sorted(normalized_str2)
```
Comment (MayMon-T3807) [inline solutions/anagram_checker.py:41]: The use of sorted for comparing normalized strings is efficient and makes the function readable.

File: solutions/tests/test_anagram_checker.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+This module contains unit tests for the `anagram_checker` function
+defined in `anagram_checker.py`. It validates the behavior for
+various input cases, including boundary and defensive checks.
+"""
+
+import unittest
+from ..anagram_checker import anagram_checker
+
+
+class TestAnagramChecker(unittest.TestCase):
```
Comment (MayMon-T3807) [inline solutions/tests/test_anagram_checker.py:13]: I think test class  covers Valid and invalid anagrams, Case insensitivity, Strings with spaces, and Defensive checks for input types.

**Review by MayMon-T3807 — APPROVED** [review]
This implementation and test coverage are well-thought-out, showing a strong understanding of best practices.
Excellent use of Python's built-in functions for readability and performance.
Overall, great job Pyae Linn for the solution with clear documentation and comprehensive testing!

**Comment by MayMon-T3807** [comment]
> I removed "MayMon-T3807" by mistake. Sorry.

That's okay. I think it works now.

**Comment by PyaeLinn01** [comment]
Thank all of you guys.
